### PR TITLE
Create Source

### PR DESCRIPTION
**Pull request by luancazarine**
Trigger:
 - New Ticket

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4080"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

**Review by feyzullah — DISMISSED** [review]

**Comment by DilanAthukorala** [comment]
@luancazarine observing following issues;

1. Please correct the type error shown below
image


2. observing following error when creating the source
image

**Comment by todsac** [comment]
I fixed the typo

File: components/estreamdesk/estreamdesk.app.mjs [hunk 1]
```diff
@@ -1,11 +1,84 @@
...
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getBaseUrl() {
+      return `${this.$auth.domain}/api`;
+    },
+    _getHeaders() {
+      return {
+        "apiKey": `${this.$auth.api_key}`,
```
Comment (lcaresia) [inline components/estreamdesk/estreamdesk.app.mjs:14]: Move this to a getter method.

```suggestion
        "apiKey": `${this._apiKey()}`,
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

I requested just some fixes.

File: components/estreamdesk/estreamdesk.app.mjs [hunk 2]
```diff
@@ -11,27 +11,24 @@ export default {
...
         ...otherConfig,
       };
 
-      return axios($ || this, config);
+      return axios($, config);
     },
-    async listTickets({
-      $, params,
-    }) {
+    async listTickets({ ...args } = {}) {
```
Comment (jcortes) [inline components/estreamdesk/estreamdesk.app.mjs:28]:
```suggestion
    async listTickets(args = {}) {
```

**Review by jcortes — CHANGES_REQUESTED** [review]
Hi @luancazarine I've just added one simple suggestion. Thanks

**Review by jcortes — APPROVED** [review]
Hi @luancazarine LGTM! Ready for QA!

**Comment by DilanAthukorala** [comment]
Verification completed and ready for release.

**Review by jcortes — APPROVED** [review]
Approved!

**Pull request merged**